### PR TITLE
Add ViT demo for ACT

### PR DESCRIPTION
**Pull request by RachelXu7**

### CODE REVIEW

File: example/auto_compression/image_classification/configs/VIT/data_reader.yml [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+      cls_label_path: ./dataset/ILSVRC2012/val_list.txt
+      transform_ops:
+        - DecodeImage:
+            to_rgb: True
+            channel_first: False
+        - ResizeImage:
+            resize_short: 256
+        - CropImage:
+            size: 224
+        - NormalizeImage:
```
Comment (yghstill) [inline example/auto_compression/image_classification/configs/VIT/data_reader.yml:44]: 只有normalize的mean和std和其他模型不同，不推荐直接依赖ppclas

Comment (RachelXu7) [reply example/auto_compression/image_classification/configs/VIT/data_reader.yml:44]: 预处理加了randomflip。。

Comment (yghstill) [reply example/auto_compression/image_classification/configs/VIT/data_reader.yml:44]: 自动压缩使用的是eval reader，无需randomflip。

File: example/auto_compression/image_classification/README.md [hunk 2]
```diff
@@ -111,7 +118,12 @@ python -m paddle.distributed.launch run.py --save_dir='./save_quant_mobilev1/' -
 ```
 多卡训练指的是将训练任务按照一定方法拆分到多个训练节点完成数据读取、前向计算、反向梯度计算等过程，并将计算出的梯度上传至服务节点。服务节点在收到所有训练节点传来的梯度后，会将梯度聚合并更新参数。最后将参数发送给训练节点，开始新一轮的训练。多卡训练一轮训练能训练```batch size * num gpus```的数据，比如单卡的```batch size```为32，单轮训练的数据量即32，而四卡训练的```batch size```为32，单轮训练的数据量为128。
 
-注意 ```learning rate``` 与 ```batch size``` 呈线性关系，这里单卡 ```batch size``` 为32，对应的 ```learning rate``` 为0.015，那么如果 ```batch size``` 减小4倍改为8，```learning rate``` 也需除以4；多卡时 ```batch size``` 为32，```learning rate``` 需乘上卡数。所以改变 ```batch size``` 或改变训练卡数都需要对应修改 ```learning rate```。
+注意：
+
+- 参数设置：```learning rate``` 与 ```batch size``` 呈线性关系，这里单卡 ```batch size``` 为32，对应的 ```learning rate``` 为0.015，那么如果 ```batch size``` 减小4倍改为8，```learning rate``` 也需除以4；多卡时 ```batch size``` 为32，```learning rate``` 需乘上卡数。所以改变 ```batch size``` 或改变训练卡数都需要对应修改 ```learning rate```。
```
Comment (ceci3) [inline example/auto_compression/image_classification/README.md:123]: > 多卡时 ```batch size``` 为32，```learning rate``` 需乘上卡数。
这句话是什么意思， 多卡运行时候，batch size设置的是单卡的bs？

File: example/auto_compression/image_classification/configs/VIT/qat_dis.yaml [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+  activation_quantize_type: moving_average_abs_max
+  weight_quantize_type: channel_wise_abs_max
+  not_quant_pattern:
+  - skip_quant
+  quantize_op_types:
+  - conv2d
+  - depthwise_conv2d
+  - matmul
+  - matmul_v2
+  weight_bits: 8
```
Comment (ceci3) [inline example/auto_compression/image_classification/configs/VIT/qat_dis.yaml]: 这些默认参数就不用写出来了

File: example/auto_compression/image_classification/configs/VIT/qat_dis.yaml [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+  model_dir: ViT_base_patch16_224_infer
+  model_filename: inference.pdmodel
+  params_filename: inference.pdiparams
+  batch_size: 16
+  input_name: inputs
+  reader_config: ./configs/VIT/data_reader.yml
+
+Distillation:
+  alpha: 1.0
+  loss: l2
```
Comment (ceci3) [inline example/auto_compression/image_classification/configs/VIT/qat_dis.yaml]: 这些默认参数也不用写了

File: example/auto_compression/image_classification/run_ppclas.py [hunk 5]
```diff
@@ -0,0 +1,170 @@
...
+                results.append(result)
+            t.update()
+    result = np.mean(np.array(results), axis=0)
+    return result[0]
+
+
+def main():
+    rank_id = paddle.distributed.get_rank()
+    if args.devices == 'gpu':
+        place = paddle.CUDAPlace(rank_id)
```
Comment (ceci3) [inline example/auto_compression/image_classification/run_ppclas.py]: place没有用到吧

File: example/auto_compression/image_classification/run_ppclas.py [hunk 6]
```diff
@@ -0,0 +1,170 @@
...
+        device = paddle.set_device('gpu')
+    else:
+        place = paddle.CPUPlace()
```
Comment (ceci3) [inline example/auto_compression/image_classification/run_ppclas.py]: 同上

File: example/auto_compression/image_classification/run_ppclas.py [hunk 7]
```diff
@@ -0,0 +1,170 @@
...
+                  dict) and all_config['TrainConfig']['learning_rate'][
+                      'type'] == 'CosineAnnealingDecay':
+        step = int(
+            math.ceil(
+                float(args.total_images) / (global_config['batch_size'] *
+                                            gpu_num)))
+        all_config['TrainConfig']['learning_rate']['T_max'] = step
+        print('total training steps:', step)
+
+    init_logger()
```
Comment (ceci3) [inline example/auto_compression/image_classification/run_ppclas.py:142]: log就不用从paddleclas import了吧

Comment (RachelXu7) [reply example/auto_compression/image_classification/run_ppclas.py:142]: 需要init 否则是None 会bug

File: example/auto_compression/image_classification/README.md (hunk 2 above)
Comment (RachelXu7) [reply example/auto_compression/image_classification/README.md:123]: 意思是 总bs和lr线性关系 不管多卡单卡 需要相应修改，之前就是这么写的，没啥问题吧

**Review by ceci3 — APPROVED** [review]
